src/guild/settings.rs

```diff
@@ -1,10 +1,60 @@
-use std::collections::HashMap;
+use std::collections::{HashMap, HashSet};
 
 use serenity::{model::id::GuildId, prelude::TypeMapKey};
 
-#[derive(Default)]
 pub struct GuildSettings {
     pub autopause: bool,
+    pub allowed_domains: HashSet<String>,
+    pub banned_domains: HashSet<String>,
+}
+
+pub trait Update {
+    fn set_allowed_domains(&mut self, allowed_str: &str);
+    fn set_banned_domains(&mut self, banned_str: &str);
+    fn update_domains(&mut self);
+}
+
+impl Default for GuildSettings {
+    fn default() -> GuildSettings {
+        GuildSettings {
+            autopause: false,
+            allowed_domains: HashSet::from(["youtube.com".to_string()]),
```

Comment:
Put the default allowed domain in a const top level variable for visibility and easy  change